### ISSUE TITLE
False positives on non-style JS/TS (not reproduced by stylelint CLI)

### ISSUE THREAD

**Issue by bensaufley** [shinnn/vscode-stylelint#258]
Scattered throughout my JS I'm getting false positives, generally to do with indentation. Example [here](https://user-images.githubusercontent.com/331287/50117417-eec7fd80-021a-11e9-8858-8d93e6e5802c.png), and tooltip shown [here](https://user-images.githubusercontent.com/331287/50117442-fe474680-021a-11e9-8adb-357bff56f121.png). [Here](https://user-images.githubusercontent.com/331287/50117911-3438fa80-021c-11e9-82a5-fd8988804340.png) is another example. Running my package script `lint:css`, which is `stylelint --color "src/**/*.{js,jsx}"`, produces none of these errors, so this appears to be specific to this VS Code plugin. Here's my `.stylelintrc.json`, at the root of my project:

```json
{
  "processors": ["stylelint-processor-styled-components"],
  "extends": [
    "stylelint-config-recommended",
    "stylelint-config-styled-components"
  ],
  "rules": {
    "block-opening-brace-space-before": "always",
    "color-hex-length": "short",
    "color-named": "never",
    "declaration-bang-space-before": "always",
    "declaration-bang-space-after": "never",
    "declaration-block-semicolon-newline-after": "always",
    "declaration-colon-space-before": "never",
    "declaration-colon-space-after": "always",
    "declaration-no-important": null,
    "function-comma-space-after": "always",
    "function-parentheses-space-inside": "never",
    "indentation": 2,
    "length-zero-no-unit": true,
    "max-nesting-depth": 5,
    "no-descending-specificity": null,
    "number-leading-zero": "never",
    "property-no-vendor-prefix": [true, {
      "ignoreProperties": ["appearance", "grid-column", "grid-columns", "grid-row", "grid-rows", "grid-row-align"]
    }],
    "selector-list-comma-newline-after": "always",
    "selector-no-qualifying-type": [true, {
      "ignore": ["class"]
    }],
    "selector-pseudo-element-colon-notation": "double",
    "shorthand-property-no-redundant-values": true,
    "value-no-vendor-prefix": [true, {
      "ignoreValues": ["grid"]
    }]
  }
}
```

Package dependencies:

```json
    "stylelint": "^9.7.1",
    "stylelint-config-recommended": "^2.1.0",
    "stylelint-config-styled-components": "^0.1.1",
    "stylelint-processor-styled-components": "^1.5.0",
```

vscode-stylelint version: `0.47.0`; VS Code version 1.29.1 MacOS. Please let me know if there's anything else I can tell you. Thanks!

**Comment by shinnn** [comment]
This extension doesn't use stylelint CLI but its API.

-----

If you still cannot reproduce problems with API, feel free to comment here.

**Comment by bensaufley** [comment]
So, @shinnn … what does the API do differently from the CLI to produce incorrect results? How can I prevent this clearly wrong behavior from occurring when it is not occurring in any other way that I invoke stylelint?

**Comment by shinnn** [comment]
https://github.com/stylelint/stylelint/issues

**Comment by shinnn** [comment]
@bensaufley Why did you delete this comment?

> So, @shinnn … what does the API do differently from the CLI to produce incorrect results? How can I prevent this clearly wrong behavior from occurring when it is not occurring in any other way that I invoke stylelint?

Event if users delete their comments, I can easily read them via [GitHub user events API](https://developer.github.com/v3/activity/events/#list-events-performed-by-a-user).

[events.json.zip](https://github.com/shinnn/vscode-stylelint/files/2688671/events.json.zip)

**Comment by bensaufley** [comment]
I hadn’t initially seen the second half of your original comment and decided the comment wasn’t necessary and my time was better spent tracking this down elsewhere. I don’t know what your point is about the events API—I assumed you got notifications one way or another but you’d see I’d retracted the comment if/when you went to respond.
